src/pages/fieldManagement/settings/defaultFieldComponents/RelationshipEditorCore.jsx

```diff
@@ -0,0 +1,239 @@
+            <InputAdornment position="end">
+              <DeleteButton
+                onClick={() => {
+                  onChange(
+                    deleteRelationship(
+                      allRelationships,
+                      currentCategory,
+                      relationship,
+                    ),
+                  );
+                }}
+              />
+            </InputAdornment>
+          ),
+        }}
+      />
+    </div>
+  );
+}
+
+function renderSingleRelationshipObj(
+  allRelationships,
+  targetKey,
+  onChange,
+) {
+  const relationships = get(allRelationships, targetKey, []);
+  return (
+    <div style={{ marginTop: 10 }} key={targetKey}>
+      <TextInput
+        width={240}
+        schema={targetKey}
+        onChange={newName => {
+          onChange(
+            updateCategoryName(allRelationships, targetKey, newName),
+          );
+        }}
+        value={targetKey}
+        autoFocus
+        InputProps={{
+          endAdornment: (
+            <InputAdornment position="end">
+              <IconButton
+                size="small"
+                onClick={() => {
+                  onChange(
+                    addRelationshipToCategory(
+                      allRelationships,
+                      targetKey,
+                      '',
+                    ),
+                  );
+                }}
+              >
+                <NewChildIcon />
+              </IconButton>
+              <DeleteButton
+                onClick={() => {
+                  onChange(
+                    deleteCategory(allRelationships, targetKey),
+                  );
+                }}
+              />
+            </InputAdornment>
+          ),
+        }}
+      />
+      {relationships.map(relationship =>
+        renderSingleRelationship(
+          allRelationships,
+          targetKey,
+          relationship,
+          onChange,
+        ),
+      )}
+    </div>
+  );
+}
+
+export default function RelationshipEditorCore({
+  schema,
+  value,
+  onChange,
+  siteSettings, // unpacking to prevent passing to div
+  ...rest
+}) {
+  const justRelationshipsFormPart = get(
+    value,
+    'relationships',
+    value,
+  );
+  return (
+    <div {...rest}>
+      <div
+        style={{
+          display: 'flex',
+          justifyContent: 'space-between',
+          alignItems: 'center',
+        }}
+      >
+        <Text variant="h5" id="RELATIONSHIP_EDITOR" />
+        <Button
+          onClick={() => {
+            onChange(createCategory(justRelationshipsFormPart));
+          }}
+          style={{ width: 200 }}
+          size="small"
+        >
+          <FormattedMessage id="NEW_RELATIONSHIP" />
+        </Button>
+      </div>
+      <div
+        style={{
+          display: 'flex',
+          flexDirection: 'column',
+          padding: 20,
+        }}
+      >
+        {Object.keys(justRelationshipsFormPart).map(key =>
+          renderSingleRelationshipObj(
```

Comment:
`renderSingleRelationshipObj` should be a component and should be called as one.
At that point, "render" is implied and "Obj" is misleading, so maybe it should be RelationshipTree or RelationshipTreeEditor.